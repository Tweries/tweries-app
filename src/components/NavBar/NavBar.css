.NavBar {
  background-color: white;
  border: 1px solid rgb(230, 236, 240);
  display: flex;
  flex-direction: column;
  align-items: center;
  max-width: 320px;
  margin-top: 16px;
  padding: 8px;
  width: 320px;
}

.NavBar img {
  border: 1px solid rgb(230, 236, 240);
}
