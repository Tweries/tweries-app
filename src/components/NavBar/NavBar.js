import React, { useEffect } from 'react';
import { useAuth0 } from '../../react-auth0-wrapper';
import './NavBar.css';

const NavBar = ({ dispatch }) => {
  const { isAuthenticated, loginWithRedirect, logout, user } = useAuth0();

  useEffect(() => {
    if (isAuthenticated) {
      dispatch({ type: 'SET_USER_ID', value: user.sub });
    }
  }, [dispatch, isAuthenticated, user]);

  return (
    <header className="NavBar">
      {isAuthenticated ? (
        [
          <img alt={user.name} key="img" src={user.picture} />,
          <span key="span">
            <span aria-label="hello" role="img">
              ðŸ‘‹
            </span>{' '}
            {user.name}
          </span>,
          <button
            data-testid="logout"
            key="button"
            onClick={() =>
              logout({
                returnTo: window.location.href
              })
            }
          >
            Log out
          </button>
        ]
      ) : (
        <button data-testid="login" onClick={() => loginWithRedirect({})}>
          Log in
        </button>
      )}
    </header>
  );
};

export default NavBar;
