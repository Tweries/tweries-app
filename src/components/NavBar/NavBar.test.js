import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { useAuth0 } from '../../react-auth0-wrapper';
import NavBar from './NavBar';

jest.mock('../../react-auth0-wrapper');

const user = {
  name: 'china-musk',
  picture:
    'https://abs.twimg.com/sticky/default_profile_images/default_profile_normal.png',
  sub: 'twitter|1183836409850814464'
};

describe('NavBar', () => {
  const scenarios = [
    {
      arrange: () => {
        useAuth0.mockImplementation(() => ({
          isAuthenticated: true,
          user
        }));
      },
      description: 'authenticated user'
    },
    {
      arrange: () => {
        useAuth0.mockImplementation(() => ({ isAuthenticated: false }));
      },
      description: 'unauthenticated user'
    }
  ];

  scenarios.forEach(({ arrange, description }) => {
    it(description, () => {
      arrange();

      const { container } = render(<NavBar dispatch={jest.fn()} />);

      expect(container).toMatchSnapshot();
    });
  });

  it('login', () => {
    const mockLoginWithRedirect = jest.fn();
    useAuth0.mockImplementation(() => ({
      isAuthenticated: false,
      loginWithRedirect: mockLoginWithRedirect
    }));

    const { getByTestId } = render(<NavBar dispatch={jest.fn()} />);
    fireEvent.click(getByTestId('login'));

    expect(mockLoginWithRedirect).toBeCalled();
  });

  it('logout', () => {
    const mockDispatch = jest.fn();
    const mockLogout = jest.fn();
    useAuth0.mockImplementation(() => ({
      isAuthenticated: true,
      logout: mockLogout,
      user
    }));

    const { getByTestId } = render(<NavBar dispatch={mockDispatch} />);
    fireEvent.click(getByTestId('logout'));

    expect(mockDispatch.mock.calls[0]).toMatchSnapshot();
    expect(mockLogout).toBeCalled();
  });
});
