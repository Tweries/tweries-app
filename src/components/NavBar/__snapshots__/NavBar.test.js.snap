// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar authenticated user 1`] = `
<div>
  <header
    class="NavBar"
  >
    <img
      alt="china-musk"
      src="https://abs.twimg.com/sticky/default_profile_images/default_profile_normal.png"
    />
    <span>
      <span
        aria-label="hello"
        role="img"
      >
        ðŸ‘‹
      </span>
       
      china-musk
    </span>
    <button
      data-testid="logout"
    >
      Log out
    </button>
  </header>
</div>
`;

exports[`NavBar logout 1`] = `
Array [
  Object {
    "type": "SET_USER_ID",
    "value": "twitter|1183836409850814464",
  },
]
`;

exports[`NavBar unauthenticated user 1`] = `
<div>
  <header
    class="NavBar"
  >
    <button
      data-testid="login"
    >
      Log in
    </button>
  </header>
</div>
`;
